common/src/test/java/org/keycloak/common/util/MultivaluedHashMapTest.java

```diff
@@ -0,0 +1,142 @@
+
+/**
+ *
+ * @author tkyjovsk
+ */
+public class MultivaluedHashMapTest {
+
+    public void equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(MultivaluedHashMap map, MultivaluedHashMap equalMap) {
+        assertTrue(String.format("MultivaluedHashMap.equalsIgnoreValueOrder() should return `true` for the same object. \nmap: %s", map),
+                map.equalsIgnoreValueOrder(map));
+        assertTrue(String.format("MultivaluedHashMap.equalsIgnoreValueOrder() should return `true` for maps that are equal. \nmap1: %s \nmap2: %s", map, equalMap),
+                map.equalsIgnoreValueOrder(equalMap));
+        assertTrue(String.format("MultivaluedHashMap.equalsIgnoreValueOrder() should return `true` for maps that are equal. \nmap1: %s \nmap2: %s", equalMap, map),
+                equalMap.equalsIgnoreValueOrder(map));
+    }
+
+    public void equalsIgnoreValueOrder_shouldReturnFalseForDifferentMaps(MultivaluedHashMap map, MultivaluedHashMap differentMap) {
+        assertFalse(String.format("MultivaluedHashMap.equalsIgnoreValueOrder() should return `false` for maps that are different. \nmap1: %s \nmap2: %s", map, differentMap),
+                map.equalsIgnoreValueOrder(differentMap));
+        assertFalse(String.format("MultivaluedHashMap.equalsIgnoreValueOrder() should return `false` for maps that are different. \nmap1: %s \nmap2: %s", differentMap, map),
+                differentMap.equalsIgnoreValueOrder(map));
+    }
+
+    @Test
+    public void testEqualsIgnoreValueOrder_exactlyEqual() {
+        MultivaluedHashMap map = new MultivaluedHashMap();
+        MultivaluedHashMap equalMap = new MultivaluedHashMap();
+        map.put(1, Arrays.asList(new Integer[]{1, 2, 3}));
+        equalMap.put(1, Arrays.asList(new Integer[]{1, 2, 3}));
+        equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(map, equalMap);
+
+        map.put(2, Arrays.asList(new Integer[]{4, 5, 6}));
+        equalMap.put(2, Arrays.asList(new Integer[]{4, 5, 6}));
+        equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(map, equalMap);
+
+        map.put("c", Arrays.asList(new String[]{"x", "y", "z"}));
+        equalMap.put("c", Arrays.asList(new String[]{"x", "y", "z"}));
+        equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(map, equalMap);
+    }
+
+    @Test
+    public void testEqualsIgnoreValueOrder_sameLengthSameValues() {
+        MultivaluedHashMap map = new MultivaluedHashMap();
+        MultivaluedHashMap equalMap = new MultivaluedHashMap();
+        map.put(1, Arrays.asList(new Integer[]{1, 2, 3}));
+        equalMap.put(1, Arrays.asList(new Integer[]{3, 2, 1}));
+        equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(map, equalMap);
+
+        map.put(2, Arrays.asList(new Integer[]{4, 5, 6}));
+        equalMap.put(2, Arrays.asList(new Integer[]{5, 6, 4}));
+        equalsIgnoreValueOrder_shouldReturnTrueForEqualMaps(map, equalMap);
+
+        map.put("c", Arrays.asList(new String[]{"x", "y", "z"}));
```

Comment:
This is not false but I'm mitigated about this test : map was first initialized like if it was a MultivaluedHashMap<Integer, Integer>...
Here it is used like if it was a MultivaluedHashMap<String, String>. Mixing types is not what the object has been created for but it is not considered as an error...
Ideally, MultivaluedHashMap should not have mixed type content and types should be provided at declaration... but here, it is just for test purpose.

Reply:
Yes, it was just for the test. But you're right, the types shouldn't be mixed. I addressed your suggestions in the next push. Thanks!